### DISCUSSION TITLE
Bulk Checkin

### DISCUSSION THREAD

**Discussion by Thorpscu** [snipe/snipe-it#9551] [Ideas]
Currently, we have bulk check in and DELETE.  I want Bulk Check in without delete.  Or to be able to look at the user and click check in then check in.. Each time I click check in then I have to research the person again..

**Comment by betomail** [comment]
Also, should be great if we have a Bulk Check in as we have the Bulk check out. Today I'm creating a form to solve this issue However it will be awesome if we have it in Snipe IT. Thanks!

**Comment by adagioajanes** [comment]
I've actually already started two different PRs for this.

#10462 works almost identically to the current bulk check out function.
#10455 works similar to the bulk audit function, but performs check ins.

I'd love to see your comments on those and to get your feedback.

**Reply by adagioajanes** [reply]
#10455 was accepted! This feature will be included in a future version of Snipe IT.
This can now be closed.

**Comment by HCurren** [comment]
While I was very excited to see that bulk check-in is being included in the upcoming Version 6, I would still think a similar check-out feature would be immensely valuable. My company uses Snipe-IT to track tools used by it's technicians. They have to check-out about 40 tools before each job, and check them back in on arrival. It seems to be one of the parts of the job that they dread the most, due to the numerous steps required. As it is currently, the steps required for this process are as follows;

1. Remove tool from tool storage
2. Hit tab to select "search by asset number" box
3. use scanner to scan barcode
4. put both items down and fill out the forms 
5. Click check in
6. Put tool into tool bag to prep for departure.
7. Repeat steps 1 through 6 forty times. 


We would love to see a simple option to fill out the Check-out fields first (Who is the item being checked out to, when will they be checked back in, notes, etc), and then scan all of the items sequentially, reducing the steps to the following;

1. Click "Check-out assets" (or whatever it would be called)
2. fill out the forms
3. Remove tool from storage
4. Scan
5. Put tool in bag
6. Repeat steps 3 through 5 forty times. 

While it may only reduce the number of steps by 50% (*only* lol), the amount of physical effort and transitioning it would require would save us hours per week, and I do not believe I am exaggerating about this.
